ecommerce/views_test.py

```diff
@@ -35,3 +46,60 @@ def test_get_products(user_drf_client, products):
     )
 
     assert_drf_json_equal(resp.json(), ProductSerializer(product).data)
+
+
+def test_get_basket(user_drf_client, user):
+    """Test the view that returns a state of Basket"""
+    basket = BasketFactory.create(user=user)
+    BasketItemFactory.create(basket=basket)
+    resp = user_drf_client.get(
+        reverse("basket-detail", kwargs={"username": user.username})
+    )
+    assert_drf_json_equal(resp.json(), BasketSerializer(basket).data)
+
+
+def test_get_basket_items(user_drf_client, user):
+    """Test the view that returns a list of BasketItems in a Basket"""
+    basket = BasketFactory.create(user=user)
+    basket_item = BasketItemFactory.create(basket=basket)
+    basket_item_2 = BasketItemFactory.create(basket=basket)
+
+    # this item belongs to another basket, and should not be in the response
+    BasketItemFactory.create()
+    resp = user_drf_client.get("/api/baskets/{}/items".format(basket.id), follow=True)
+    basket_info = resp.json()
+    assert len(basket_info) == 2
+    assert_drf_json_equal(
+        resp.json(),
+        [
+            BasketItemSerializer(basket_item).data,
+            BasketItemSerializer(basket_item_2).data,
+        ],
+    )
+
+
+def test_delete_basket_item(user_drf_client, user):
+    """Test the view to delete item from the basket"""
+    basket_item = BasketItemFactory.create(basket__user=user)
+    basket = basket_item.basket
+    assert basket.basket_items.count() == 1
+    user_drf_client.delete(
+        "/api/baskets/{}/items/{}/".format(basket.id, basket_item.id),
```

Comment:
Should the first part of the url be `username` so that it aligns to the `/api/baskets/:username/` API?

Reply:
yeah, I find it interesting that even though we set the look up key to username the lookup by pk is still working.